packages/next/src/server/lib/clock.ts

```diff
@@ -0,0 +1,16 @@
+export function getCurrentTimestamp() {
+  if (process.env.NEXT_RUNTIME === 'edge') {
+    if (performance.timeOrigin === undefined) {
+      return Date.now()
+    } else {
+      return Math.floor(performance.timeOrigin + performance.now())
+    }
+  } else {
+    const performance = require('node:perf_hooks').performance
```

Comment:
Is there any version of Node.js that we support that doesn't have it on the global `performance.timeOrigin`?

Reply:
no, updated

Reply:
I believe there are some issues with missing performance hooks in edge so we might need to double check there https://github.com/vercel/next.js/pull/64669

Reply:
`measure` is a much newer API than the ones we need here.